### ISSUE TITLE
False positive extending both Enum and another class

### ISSUE THREAD

**Issue by jfmend** [microsoft/pyright#11126]
False positive extending both Enum and another class:

```python
from enum import Enum
from typing import NamedTuple

class C(NamedTuple):
    n: int

class E(C, Enum):
    c1 = C(1)  # <- Pyright error on the C constructor call
    # Error message:
    #   Argument of type "C" cannot be assigned to parameter "n" of type "int" in function "__new__"
    #       "C" is not assignable to "int"

# In runtime, everything seems to work fine:
print(E.c1.n)  # -> 1
print(E.c1.name)  # -> 'c1'
```
The error remains if instead of being a ```NamedTuple```, C has a simple ```__init__``` method assigning ```self.n```

Tested on Python v3.14 with Pylance v2025.9.1

**Comment by pbchekin** [comment]
I have the same issue with a class that inherits both Enum and NamedTuple, slightly different error message:

```python
import enum
from typing import NamedTuple

class Item(NamedTuple):
    a: str
    b: str

class Items(Item, enum.Enum):
    I = Item(a='a', b='b')

print(Items.I.a, Items.I.b)
```

The error message is `error: Argument missing for parameter "b" (reportCallIssue)`. Note that the error is about parameter `b` but not `a`.

Tested on Python 3.12 with Pyright 1.1.407.

**Comment by pbchekin** [comment]
While runtime works with multiple inheritance of NamedTuple and Enum it is officially not supported:

> NamedTuple does not support multiple inheritance. Type checkers should enforce this limitation:

See https://typing.python.org/en/latest/spec/namedtuples.html
